### DISCUSSION TITLE
How to solve error 'Unhandled method window/workDoneProgress/create'

### DISCUSSION THREAD

**Discussion by newfish-cmyk** [TypeFox/monaco-languageclient#542] [Q&A]
I tried to replace pyright with typescript-language-server in the example code, but get errors
![image](https://github.com/TypeFox/monaco-languageclient/assets/71265218/9d9aa1ea-e4c5-43cd-80c0-91c7bfc39e05)
After doing some research, I found that 

> To keep the protocol backwards compatible servers are only allowed to use window/workDoneProgress/create request if the client signals corresponding support using the client capability window.workDoneProgress

However, I noticed that the original python example also has `"workDoneProgress": "true"` in the initialization. So why doesn't the python example cause errors, while using typescript-language-server does? 
Any help troubleshooting would be great!

**Comment by kaisalmen** [comment]
Hi @newfish-cmyk can you share a repo with dependencies or a fork of this repo with the changed code, something we can build/use?

**Reply by newfish-cmyk** [reply]
https://github.com/newfish-cmyk/monaco-languageclient/tree/ts 

here I simply installed `typescript-language-server`, and changed `examples/src/python/server/direct.ts`

and then, you can run the examples,  click 'python'

**Comment by newfish-cmyk** [comment]
So a friend told me how to fix this:

You just gotta go into the file ` monaco-languageclient/node_modules/vscode-jsonrpc/lib/common/connection.js ` and delete line 540, which is:

```
replyError(new messages_1.ResponseError(messages_1.ErrorCodes.MethodNotFound, `Unhandled method ${requestMessage.method}`), requestMessage.method, startTime);
```

After that, the example starts working!

Seems like there's some kinda conflict between the client initialization and the websocket connection, but I still don't really get how to properly fix this for good. 😖

**Reply by kaisalmen** [reply]
@newfish-cmyk I haven't had time to look into this. New version of this lib is currently in the making.

**Reply by kaisalmen** [reply]
@newfish-cmyk the server sends `"method":"window/workDoneProgress/create"` directly and the client can't handle it and sends the error leading the server to crash. I don't really know `typescript-language-server`, so I can't help you there. I tried a couple of things, but was not able to prevent it from crashing. You could look at other client implementations (if there are ones, I did not search) or try another TS language server.

**Reply by newfish-cmyk** [reply]
thank you for your help and advice, I will give it a try

**Reply by tmisirpash** [reply]
We are also hitting this in one of our projects (specifically with typescript-language-server). Do you have any recommendations on where we can look to fix this, @kaisalmen ?
